[Song Si Hyun] iP

hyxnnii opened the pull request:
### Kira

> “Very demure, very mindful, very cutesy.”

Kira is your **personal secretary** that is not just cute, but keeps track of all the overwhelming things you need to do🤭

Kira is:
- text-based
- easy to learn
- SUPER FAST to use

Kira can:
1. Add tasks
2. `Mark` tasks as done
3. `Prioritise` tasks
4. `Delete` tasks
5. View tasks
6. `Find` tasks

So what's in your task list right now?

```
You have 1 task in your list:
```
- [ ] Download Kira

Click [here](https://hyxnnii.github.io/ip/) for user guide

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

hyxnnii committed:
Add level 0: rename, greet, exit

hyxnnii committed:
Add level 1: echo

hyxnnii committed:
Add level 2: add, list

hyxnnii committed:
Add level 3: mark as done

hyxnnii committed:
Add level 4: todo, deadline, event

hyxnnii committed:
Add A-TextUiTesting

hyxnnii committed:
Add level-5: handle errors

hyxnnii committed:
Add level-6: delete

hyxnnii committed:
Add level-6: delete

hyxnnii committed:
Add level-7: partially done

hyxnnii committed:
Add level-7: save

GuanpengR commented on src/main/java/Task.java:
```diff
@@ -0,0 +1,44 @@
+public class Task {
+
+    private String input;
+    private boolean done;
```
Perhaps naming this boolean to isDone

GuanpengR commented on src/main/java/List.java:
```diff
@@ -0,0 +1,52 @@
+            return "Now you have " + tasks.size() + " tasks in the list.\n";
+        }
+    }
+    public void addTaskToList(Task task) {
```
Clear and appropriate name!

GuanpengR commented on src/main/java/Kira.java:
Perhaps consider including all classes into packages

GuanpengR commented on src/main/java/List.java:
```diff
@@ -0,0 +1,52 @@
+    String line = "____________________________________________________________\n";
+
+    public List(){
+        this.tasks = new ArrayList<>(100);
```
Nice use of plural to indicate collection of tasks in the list.

hyxnnii committed:
Add level-8: date

hyxnnii committed:
Add level-8: under correct branch

hyxnnii committed:
Add level-8: under correct branch

hyxnnii committed:
Add level-8: under correct branch

hyxnnii committed:
Add A-MoreOOP: parser, storage, ui class

hyxnnii committed:
Add A-MoreOOP: parser, storage, ui class

hyxnnii committed:
Merge branch 'branch-Level-8'

* branch-Level-8:
  Add level-8: under correct branch
  Add level-8: under correct branch
  Add level-8: under correct branch
  Add level-8: date
  Add level-7: save
  Add level-7: partially done

hyxnnii committed:
Merge branch 'branch-A-MoreOOP'

* branch-A-MoreOOP:
  Add A-MoreOOP: parser, storage, ui class
  Add A-MoreOOP: parser, storage, ui class

hyxnnii committed:
Add A-Packages: organise into packages

hyxnnii committed:
Add A-Packages: correct branch

hyxnnii committed:
Merge branch 'branch-A-Packages'

* branch-A-Packages:
  Add A-Packages: correct branch

hyxnnii committed:
Merge commit '81a9c537afe8492675eb253968939364d65db091'

* commit '81a9c537afe8492675eb253968939364d65db091':
  build.gradle: Prevent generating a second JAR file
  Bump gradle and lib version
  Add Gradle support

hyxnnii committed:
Add A-Gradle: set up gradle

hyxnnii committed:
Add A-Gradle: correct branch

hyxnnii committed:
Add A-Gradle: correct branch

hyxnnii committed:
Merge branch 'add-gradle-support'

* add-gradle-support:
  Add A-Gradle: correct branch

hyxnnii committed:
Add A-JUnit: JUnit tests

hyxnnii committed:
Merge branch 'branch-A-JUnit'

* branch-A-JUnit:
  Add A-JUnit: JUnit tests

hyxnnii committed:
Add A-JUnit: JUnit tests

hyxnnii committed:
Merge branch 'branch-A-Jar'

* branch-A-Jar:
  Add A-JUnit: JUnit tests

hyxnnii committed:
Add A-JUnit: JUnit tests

hyxnnii committed:
Add A-JUnit: JUnit tests

hyxnnii committed:
Add A-JUnit: JUnit tests

hyxnnii committed:
Add A-JUnit: JUnit tests

hyxnnii committed:
Add A-JUnit: JUnit tests

hyxnnii committed:
Add Level-9: Find

hyxnnii committed:
resolved conflict for branch-A-JavaDoc

hyxnnii committed:
Merge branch 'branch-A-CodingStandard'

* branch-A-CodingStandard:
  Add A-JUnit: JUnit tests

hyxnnii committed:
resolved conflict for branch-Level-9

prave1n commented on src/main/java/Exceptions/UnreadableException.java:
```diff
@@ -0,0 +1,10 @@
+
+    public UnreadableException() {
+        super("____________________________________________________________\n"
+                + "OOPS!!! I'm sorry, but I don't know what that means :-(\n"
```
Good job starting the new line with "+"

prave1n commented on src/main/java/Kira/Parser.java:
```diff
@@ -0,0 +1,110 @@
+import Exceptions.EmptyException;
+import Exceptions.InvalidTaskException;
+import Exceptions.UnreadableException;
+import Tasks.*;
```
Avoid wildcard imports as imported classes should always be listed explicitly

prave1n commented on src/main/java/Kira/Parser.java:
```diff
@@ -0,0 +1,110 @@
+import Tasks.*;
+
+import java.util.Objects;
+
```
Javadoc header comment is required for classes

prave1n commented on src/main/java/Tasks/Task.java:
```diff
@@ -0,0 +1,141 @@
+public abstract class Task {
+
+    private String input;
+    private boolean done;
```
```suggestion
    private boolean isDone;
```

boolean could be named isDone to follow the naming standards

prave1n reviewed (commented):
Looks good. Consider including Javadoc header comments for classes as well

hyxnnii committed:
Level-10: GUI (partial)

hyxnnii committed:
Merge branch 'branch-Level-10'

* branch-Level-10:
  Level-10: GUI (partial)

hyxnnii committed:
Level-10: GUI

hyxnnii committed:
Merge branch 'branch-Level-10'

* branch-Level-10:
  Level-10: GUI

hyxnnii committed:
Create JAR file

hyxnnii committed:
Make error messages visible in GUI

Error messages from getResponse() need to be shown.

Error messages need to be returned instead of being printed out.

hyxnnii committed:
Add assertions

Assertions are needed to document important assumptions
that should hold at various points in the code.

hyxnnii committed:
Improve coding standard

Some parts of code do not adhere to SLAP.

Adding more abstraction improves SLAP and
makes the code more readable.

hyxnnii committed:
Improve coding standard

There are some feedback made on improving coding standards.

Renaming packages, reorganising packages, renaming some
variables allow for better adherence to coding standards.

hyxnnii committed:
Merge pull request #2 from hyxnnii/branch-A-Assertions

Branch a assertions

hyxnnii committed:
Resolve conflict between branch-A-CodeQuality and branch-A-Assertion

There are minor conflicts between the two branches.

Careful resolution of conflict required to retain improvement of code
made in both branches.

hyxnnii committed:
Merge pull request #3 from hyxnnii/branch-A-CodeQuality

Branch a code quality

hyxnnii committed:
Modify photo for kira bot and user

User photo is missing and kira photo could be better.

Add the user photo and change kira photo such that they
complement each other.

hyxnnii committed:
Add priority extension

Setting priority of tasks is a useful feature for users.

Users are able to use commands "prioritise" and "unprioritise"
to set priority of tasks and priority of tasks are displayable.

hyxnnii committed:
Merge branch 'branch-C-Priority'

* branch-C-Priority:
  Add priority extension

hyxnnii committed:
Fix MainWindow layout

MainWindow does not adjust according to size of tab.

Modify the fxml file such that text box, send button and scroll
pane adjust according to tab size.

hyxnnii committed:
Add product name to title of window

Application window is less identifiable to be Kira without any label.

Product name 'Kira' is added to the title of the window using GUI.

This makes the application more identifiable and stylised.

hyxnnii committed:
Add README page

A README page introduces the application to possible users.

Product description, feature descriptions and examples, and
application screenshot are added.

The README page needs to be comprehensive to allow users to be able
to use the application easily.

hyxnnii committed:
Merge branch 'branch-A-UserGuide'

* branch-A-UserGuide:
  Add README page

hyxnnii committed:
Change product screenshot

Product screenshot has a lot of empty spaces.

New product screenshot is taken, with more tasks added and featuring
list function.

Product screenshot needs to be more filled to look more functional
and appealing to potential users reading the readme.

hyxnnii committed:
Revise unit testings and add more javadoc comments

Unit tests are not working correctly as they are outdated. More javadoc
comments are necessary to make the code more readable.

Unit tests are updated with up-to-date parameters and results, and more
unit tests are added for greater robustness. More javadoc comments are added
and some are revised.

hyxnnii committed:
Unnecessary codes are deleted

Having unnecessary codes makes the code overall less readable and more
confusing.

Deleting methods that are unused and irrelevant comments makes the code
cleaner.
